### ISSUE TITLE
[BUG] net.Dial() = dial tcp connect: operation timed out

### ISSUE THREAD

**Issue by sparksparker** [code-ready/crc#2730]
### General information

  * OS: Linux / macOS / Windows - **macOS**
  * Hypervisor: KVM / Hyper-V / hyperkit - **hyperkit**
  * Did you run `crc setup` before starting it (Yes/No)? -  **Yes**
  * Running CRC on: Laptop / Baremetal-Server / VM - **Macbook Pro 2.6 GHz 6-Core Intel Core i7, 16 GB memory**

## CRC version
```bash
# Put `crc version` output here
CodeReady Containers version: 1.32.1+4d40571f
OpenShift version: 4.8.5 (bundle installed at /Applications/CodeReady Containers.app/Contents/Resources/crc_hyperkit_4.8.5.crcbundle)
```
  
## CRC status
```bash
# Put `crc status --log-level debug` output here
DEBU CodeReady Containers version: 1.32.1+4d40571f
DEBU OpenShift version: 4.8.5 (bundle installed at /Applications/CodeReady Containers.app/Contents/Resources/crc_hyperkit_4.8.5.crcbundle)
DEBU Running 'crc status'
DEBU Checking file: /Users/spark/.crc/machines/crc/.crc-exist
DEBU Checking file: /Users/spark/.crc/machines/crc/.crc-exist
DEBU Found binary path at /Applications/CodeReady Containers.app/Contents/Resources/crc-driver-hyperkit
DEBU Launching plugin server for driver hyperkit
DEBU Plugin server listening at address 127.0.0.1:52397
DEBU () Calling .GetVersion
DEBU Using API Version 1
DEBU () Calling .SetConfigRaw
DEBU () Calling .GetMachineName
DEBU (crc) Calling .GetState
DEBU (crc) Calling .GetBundleName
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1
DEBU Using ssh private keys: [/Users/spark/.crc/machines/crc/id_ecdsa /Users/spark/.crc/cache/crc_hyperkit_4.8.5/id_ecdsa_crc]
DEBU SSH command results: err: <nil>, output: 32737570816 15120437248 /sysroot
DEBU Making call to close driver server
DEBU (crc) Calling .Close
DEBU Successfully made call to close driver server
DEBU Making call to close connection to plugin binary
CRC VM:          Running
OpenShift:       Running (v4.8.5)
Disk Usage:      15.12GB of 32.74GB (Inside the CRC VM)
Cache Usage:     12.93GB
Cache Directory: /Users/spark/.crc/cache
```

## CRC config
```bash
# Put `crc config view` output here
- consent-telemetry          : no
```

## Host Operating System
```bash
# Put the output of `sw_vers` in case of Mac
ProductName:	macOS
ProductVersion:	11.5.2
BuildVersion:	20G95
```

### Steps to reproduce

  1. crc setup
  2. crc start

### Expected
To complete without errors

### Actual
**In the tray's logs:**
```
assigning xxx.xxx.xxx.x/xx to
net.Dial() = dial tcp xx.xxx.x.x:xxx: connect: operation timed out
...
...
net.Dial() = dial tcp xxx.xxx.xxx.x:xxxxx: connect: operation timed out
...
...
```
OpenShift stops and becomes unreachable
Unable to complete [3.3.3. Accessing the internal OpenShift registry](https://access.redhat.com/documentation/en-us/red_hat_codeready_containers/1.32/html/getting_started_guide/using-codeready-containers_gsg#accessing-the-internal-openshift-registry_gsg)

### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

[crc-start_log-level-debug](https://gist.github.com/sparksparker/7ed4b8add88448eff20c841dbb966814)

**Comment by praveenkumar** [comment]
@sparksparker As per shared logs and from issue details looks like cluster is started, what action you performed after it started which made it unreachable? Or it just happened automatic? Can you also share `crc start --log-level debug` after it become unreachable?

**Comment by sparksparker** [comment]
in addition,

on step 6 of [3.3.3. Accessing the internal OpenShift registry](https://access.redhat.com/documentation/en-us/red_hat_codeready_containers/1.32/html/getting_started_guide/using-codeready-containers_gsg#accessing-the-internal-openshift-registry_gsg),

i am getting "Error: error pushing image .... dial tcp xxx.xxx.xxx.x:xx: connect: connection refused"

**Comment by sparksparker** [comment]
> @sparksparker As per shared logs and from issue details looks like cluster is started, what action you performed after it started which made it unreachable? Or it just happened automatic? Can you also share `crc start --log-level debug` after it become unreachable?

i did not do anything for it to become unreachable. i just started it and it is currently still running. i'm not quite sure how to replicate it to become unreachable but i will update once it does

however, i am still getting the "net.Dial() = dial tcp ..... operation timed out" as we speak, along with my comment above

**Comment by praveenkumar** [comment]
> however, i am still getting the "net.Dial() = dial tcp ..... operation timed out" as we speak, along with my comment above

@sparksparker This is debug log, you can ignore right now. Also please put all the steps which you used and make this cluster unreachable once you able to reproduce it.

**Comment by gbraad** [comment]
> net.Dial() = dial tcp xxx.xxx.xxx.x:xxxxx: connect: operation timed out

when these occur during startup, it is mostly due to slow start. these are expected during the IP address assignment. however, as soon as the SSH runner is able to setup the environment, this is most likely not the reason.

I do see you have a modest macbook with 16GB of memory and you have not changed any assignment. Are you also running something else at the same time? 16G is really the bare minimum setup ...

**Comment by gbraad** [comment]
> OpenShift stops and becomes unreachable

Since the startup finished as expected, but becomes unreachable over time, we need to understand which steps you perform afterwards. Is this during the process of pushing an image to the registry?

**Comment by sparksparker** [comment]
seems to work fine now, but have another issue that i'll open in another thread. thanks for your help!
